src/lexer/Lexer.ts

```diff
@@ -83,7 +83,7 @@ export class Lexer {
      * @param options options used to customize the scan process
      * @returns an object containing an array of `errors` and an array of `tokens` to be passed to a parser.
      */
-    public scan(toScan: string, options?: ScanOptions): Lexer {
+    public scan(toScan: string, options?: ScanOptions): this {
```

Comment:
Oh, interesting. I didn't know this was a thing.

Reply:
I didn't either, until the linter complained about it. I like that because it helps show which functions can be chained.